assets/js/components/SpinnerButton.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CircularProgress } from '../material-components';
+import Button from './Button';
+
+export default function SpinnerButton( props ) {
+	const { className, onClick = () => {}, ...restProps } = props;
```

Comment:
Can you destructure the props from the function itself and add `defaultProps` instead of having a default value? Please refer to my comments below.
```suggestion
export default function SpinnerButton( { className, onClick, ...restProps } ) {
```

Reply:
> Can you destructure the props from the function itself ...

This makes it harder to read the function definition in my opinion. 

> add defaultProps instead of having a default value

As answered in another comment, the default props will be deprecated at some point in the future. The correct way to define default values is to set it when destructing `props`.